### PR TITLE
API invocation improvements

### PR DESCRIPTION
**Pull request by kylelaker-ccpo**
- Remove ANY / route
- Enable tracing for REST API by default
- Add a way to create users for the API

### CODE REVIEW

File: lib/atat-web-api-stack.ts [hunk 1]
```diff
@@ -32,7 +33,10 @@ export class AtatWebApiStack extends cdk.Stack {
     }
 
     const api = new AtatRestApi(this, "HothApi", apiProps);
-    api.restApi.root.addMethod("ANY");
+    const readUser = new ApiUser(this, "ReadUser", { secretPrefix: "api/user/snow", username: "ReadUser" });
+    const writeUser = new ApiUser(this, "ReadUser", { secretPrefix: "api/user/snow", username: "ReadUser" });
```
Comment (kylelaker-ccpo) [inline lib/atat-web-api-stack.ts:37]:
```suggestion
    const writeUser = new ApiUser(this, "ReadUser", { secretPrefix: "api/user/snow", username: "WriteUser" });
```

File: lib/constructs/api-user.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  constructor(scope: Construct, id: string, props: ApiUserProps) {
+    super(scope, id);
+
+    const user = new iam.User(this, `ApiUser${props.username}`, { path: "api-user" });
+    const accessKey = new iam.AccessKey(this, `ApiUser${props.username}Key`, {
+      user,
+    });
+    const secret = new secrets.Secret(this, `ApiUser${props.username}KeySecret`, {
+      secretName: `${props.secretPrefix}/${props.username}`,
+      secretStringValue: cdk.SecretValue.resourceAttribute(
```
[the remark this answers is not in the document]
Comment (kylelaker-ccpo) [reply lib/constructs/api-user.ts:41]: The more I think about this, the more I like the second option.

File: lib/atat-web-api-stack.ts [hunk 3]
```diff
@@ -32,7 +33,10 @@ export class AtatWebApiStack extends cdk.Stack {
     }
 
     const api = new AtatRestApi(this, "HothApi", apiProps);
-    api.restApi.root.addMethod("ANY");
```
[the remark this answers is not in the document]
Comment (zachclark-ccpo) [reply lib/atat-web-api-stack.ts:35]:
```
 ❌  At7323WebApi failed: Error: The stack named At7323WebApi failed to deploy: UPDATE_ROLLBACK_COMPLETE: The specified value for path is invalid. It must begin and end with / and contain only alphanumeric characters and/or / characters. (Service: AmazonIdentityManagement; Status Code: 400; Error Code: ValidationError; Request ID: 01d9dcc7-a0a4-4eca-b1d3-6a1ee36e3ec2; Proxy: null), Failed to fetch resource hooks to invoke for the resource
    at prepareAndExecuteChangeSet (/home/zclark/.nvm/versions/node/v16.13.1/lib/node_modules/aws-cdk/lib/api/deploy-stack.ts:382:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at CdkToolkit.deploy (/home/zclark/.nvm/versions/node/v16.13.1/lib/node_modules/aws-cdk/lib/cdk-toolkit.ts:209:24)
    at initCommandLine (/home/zclark/.nvm/versions/node/v16.13.1/lib/node_modules/aws-cdk/lib/cli.ts:341:12)
```
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway-readme.html#breaking-up-methods-and-resources-across-stacks

Looking at the docs, `api.root.addMethod('ANY');` looks like it needs to be included if we aren't defining any actual endpoints. 
Maybe we could include it in `apigateway.ts`?

[the remark this answers is not in the document]
Comment (zachclark-ccpo) [reply lib/atat-web-api-stack.ts:35]:
```
 ❌  At7323WebApi failed: Error: The stack named At7323WebApi failed to deploy: UPDATE_ROLLBACK_COMPLETE: The specified value for path is invalid. It must begin and end with / and contain only alphanumeric characters and/or / characters. (Service: AmazonIdentityManagement; Status Code: 400; Error Code: ValidationError; Request ID: 01d9dcc7-a0a4-4eca-b1d3-6a1ee36e3ec2; Proxy: null), Failed to fetch resource hooks to invoke for the resource
    at prepareAndExecuteChangeSet (/home/zclark/.nvm/versions/node/v16.13.1/lib/node_modules/aws-cdk/lib/api/deploy-stack.ts:382:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at CdkToolkit.deploy (/home/zclark/.nvm/versions/node/v16.13.1/lib/node_modules/aws-cdk/lib/cdk-toolkit.ts:209:24)
    at initCommandLine (/home/zclark/.nvm/versions/node/v16.13.1/lib/node_modules/aws-cdk/lib/cli.ts:341:12)
```
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway-readme.html#breaking-up-methods-and-resources-across-stacks

Looking at the docs, `api.root.addMethod('ANY');` looks like it needs to be included if we aren't defining any actual endpoints. 
Maybe we could include it in `apigateway.ts`?

**Pull request closed without merging**